backend/hitas/models/housing_company.py

```diff
@@ -95,6 +114,29 @@ def total_shares_count(self) -> int:
             .aggregate(sum_shares_count=Sum("shares_count"))["sum_shares_count"]
         )
 
+    @property
+    def over_thirty_years_old(self) -> bool:
+        from hitas.models import Apartment
+
+        completion_dates: list[datetime.date | None] = Apartment.objects.filter(
+            building__real_estate__housing_company=self
+        ).values_list("completion_date", flat=True)
+
+        if None in completion_dates:
+            return False
+
+        return relativedelta(timezone.now().date(), max(completion_dates)).years >= 30
+
+    @property
+    def completed(self) -> bool:
+        from hitas.models import Apartment
+
+        completion_dates: list[datetime.date | None] = Apartment.objects.filter(
+            building__real_estate__housing_company=self
+        ).values_list("completion_date", flat=True)
+
+        return None not in completion_dates
```

Comment:
Here the same thing as above, we don't need to do the checking in python, we can simply:
```py
def is_completed(self) -> bool:
    completion_date = ...
    return bool(completion_date)
```

Reply:
Changed.